src/webui/api/torrentscontroller.cpp

```diff
@@ -124,6 +130,80 @@ namespace
             }
         }
     }
+
+    QVariantList getStickyTrackers(const BitTorrent::TorrentHandle *torrent)
+    {
+        const QString working {QLatin1String("Working")};
+        const QString disabled {QLatin1String("Disabled")};
+        const QString privateMsg {QLatin1String("This torrent is private")};
```

Comment:
Seems these strings need to be translated, try `QCoreApplication::translate("TrackerListWidget", "This torrent is private");` see if it works.

Reply:
The translation works now.

Reply:
Hell, I really don't like the tendency of recent changes to return translated text from an API (and generally, return anything visible directly to the user). You generate unnecessary dependencies again. But I guess you all don't care...

Reply:
@Piccirello 
Or maybe return some constant (integers?) here and do the translation at client side? However I'm not sure how to fetch translated strings at javascript... or this need some html tricks...

Reply:
>However I'm not sure how to fetch translated strings at javascript...

JavaScript sources are translatable as well. You can easily create error code to description text mapping there.

Reply:
>create error code to description text mapping

Not only **error** codes. Any kind of **status** codes.
>Or maybe return some constant (integers?) 

Integers aren't required, of course. They can be some string codes (e.g. "working", "paused" etc.).

Reply:
I’m having trouble following, what is the actual recommendation here? Unless someone feels very strongly, I’m inclined to leave these as is.

Reply:
The point is that the API is a data source and it should provide data in the form of their values, not their specific representation. Client app handles received data and provide it to the user in form of particular representation (e.g. API provide tracker status as [working; disabled] and then client app show it as translated string, or some color, or icon etc.).

Reply:
Wait, all of the APIs in `torrentscontroller` already return translated strings. This is a non-issue.

I will change the API back to return translated strings.

Reply:
>Wait, all of the APIs in torrentscontroller already return translated strings. This is a non-issue.

Damn! This is an old issue.
>I will change the API back to return translated strings.

Why not just fix old ones? We need to learn from other people's mistakes, not repeat them.

Well, if everyone else thinks my arguments are worthless, you can go on as you like. @Chocobo1, @zeule, @thalieht?

Reply:
You really like jumping on my PRs to report your issues with the general qbittorent architecture. Please, open a separate PR fixing the issues that you see. It should not hold back new features that follow the existing architecture.

Reply:
>Why not just fix old ones? We need to learn from other people's mistakes, not repeat them.

But changing it now will surely break API... We should enforce correct design for new code, but the issue here is old, personally I would turn a blind eye for this specific case.

@Piccirello 
Just asking for your opinion here, what do you think about the idea (do i18n strings replacement in HTML file) in https://github.com/qbittorrent/qBittorrent/pull/9375#issuecomment-421228476, is it much hassle to do it?

Reply:
It's easy to do for static strings in HTML files, but more difficult to do for api responses. We end up with code that's very tightly coupled to specific api versions, such as

```
var status;
switch (tracker.status) {
    case "Working":
        status = "QBT_TR(Working)QBT_TR[CONTEXT=TrackerListWidget]";
        break;
    case "Disabled":
        status = "QBT_TR(Disabled)QBT_TR[CONTEXT=TrackerListWidget]";
        break;
    case "This torrent is private":
        status = "QBT_TR(This torrent is private)QBT_TR[CONTEXT=TrackerListWidget]";
        break;
    default:
        status = tracker.status;
}
```

I would think it's considered a feature that the current APIs return translated string values.

Reply:
>It's easy to do for static strings in HTML files, but more difficult to do for api responses. 

Thank you, I hope that doesn't mean impossible for api responses.

>I would think it's considered a feature that the current APIs return translated string values.

I think it *might* be a feature if there are 2 seperate values returned, such as:
```
tracker.status = "working"
tracker.statusI18N = "<translated string>"
```
My point is, if we always return translated strings, then it is cumbersome to write code for the API, as one will need to take locale settings into consideration.

Reply:
>I would think it's considered a feature that the current APIs return translated string values.

Disagree.
We should not spawn such flaws where we have the ability to get around it without breaking API. And we have to seriously consider the need to get rid of this "feature" with the next more or less big API update.

In this particular case let it be as is.

>You really like jumping on my PRs to report your issues with the general qbittorent architecture. Please, open a separate PR fixing the issues that you see. It should not hold back new features that follow the existing architecture.

It's nothing personal. This is our usual practice.
Unfortunately, our resources are too limited to allow us to make major architectural changes at a time. None of the project members knows exactly all the code (especially if it has a long history). We often detect some problems in some parts of the code affected by some (possible unrelated) changes. And we try to solve them along the way, if possible (usually offer to do it to the author of the affected PR).

Reply:
Let's create issues for these large architectural changes, rather than holding up individual features. In certain cases it might be reasonable, but there's no point in holding back a new feature because we can't decide if the API should return translated strings. Just one example, but the point being that we can still make incremental progress while acknowledging large changes that need to occur.